### DISCUSSION TITLE
How to distribute .NET 8 runtime with MSIX?

### DISCUSSION THREAD

**Discussion by DrusTheAxe** [dotnet/runtime#100880] [General]
If I have an MSIX packaged app is there a .NET 8 MSIX framework package it can use?

.NET runtime self-contained has pluses but also negatives.

**Comment by amarzavery** [comment]
Any update on the question asked?

**Comment by teo-tsirpanis** [comment]
@MichaelSimons, @NikolaMilosavljevic

**Comment by NikolaMilosavljevic** [comment]
@DrusTheAxe I'm not aware of any MSIX-based solutions. Visual Studio has support for MSI-based deployments. We also support ClickOnce deployments in both Visual Studio and outside (https://github.com/dotnet/deployment-tools/tree/main/docs/dotnet-mage).

@ashnaga

**Comment by zachcwillson** [comment]
Hey folks - the desire is to be able to publish an AppX/MSIX package either to the MS Store or via download link that is not self-contained. Specifically, we need to be able to define net 8 runtime as a dependency, similar to the ability to define `Microsoft.VCLibs.140.00` as a dependency.

**Comment by ashnaga** [comment]
If desired to package as MSIX, then option is to use self-contained deployment only.

**Comment by amarzavery** [comment]
The self contained option bloats the size of the MSIX. Customers do not like that approach. There are many users who have asked for this, time and again. It would be great if .net core 8.0 can be defined as a dependency and is installed/updated automatically as part of the MSIX installation.

**Comment by huoyaoyuan** [comment]
#51574
There is a long-standing request for such a solution.
